### ISSUE TITLE
[BUG]: "could not read Password" error in `checkout` task when `fetchFilter` is provided

### ISSUE THREAD

**Issue by afscrome** [microsoft/azure-pipelines-agent#4860]
### What happened?

If you add a `fetchFilter` (from #4693) to the checkout task (as well as enabling the `AGENT_USE_FETCH_FILTER_IN_CHECKOUT_TASK` knob), then the checkout manages to fetch objects from the remotes, but fails on the final `git checkout` task with an authentication issue:

There is a warning in the result suggesting changing `fetchDepth`, however I get the same issue regardless of whether I set fetch depth

Repo pipeline:

```yaml
job: demo
displayName: 🏗️ Build
steps:
  - checkout: self
    fetchFilter: tree:0
    #fetchDepth: 0
```

### Versions

Current agent version: '3.241.0'

Operating System
Ubuntu
22.04.4
LTS

Runner Image
Image: ubuntu-22.04
Version: 20240616.1.0
Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20240616.1/images/ubuntu/Ubuntu2204-Readme.md
Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20240616.1

Runner Image Provisioner
2.0.370.1
Current image version: '20240616.1.0'

### Environment type (Please select at least one enviroment where you face this issue)

- [ ] Self-Hosted
- [X] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Ubuntu 22.04.04 LTS

### Version controll system

Git

### Relevant log output

```shell
Starting: Checkout REPO@main to s
==============================================================================
Task         : Get sources
Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
Version      : 1.0.0
Author       : Microsoft
Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=798199)
==============================================================================
Syncing repository: REPO (Git)
git version
git version 2.45.2
git lfs version
git-lfs/3.5.1 (GitHub; linux amd64; go 1.21.8)
git init "/home/vsts/work/1/s"
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint:
hint: 	git config --global init.defaultBranch <name>
hint:
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint:
hint: 	git branch -m <name>
Initialized empty Git repository in /home/vsts/work/1/s/.git/
git remote add origin https://ORG@dev.azure.com/ORG/PROJECT/_git/REPO
git config gc.auto 0
git config core.longpaths true
git config --get-all http.https://ORG@dev.azure.com/ORG/PROJECT/_git/REPO.extraheader
git config --get-all http.extraheader
git config --get-regexp .*extraheader
git config --get-all http.proxy
git config http.version HTTP/1.1
git --config-env=http.extraheader=env_var_http.extraheader fetch --force --tags --prune --prune-tags --progress --no-recurse-submodules origin --depth=1 --filter=tree:0 +edb49c62b56ceb393542ebb413c58b39130fd07e:refs/remotes/origin/edb49c62b56ceb393542ebb413c58b39130fd07e
remote: Azure Repos        
remote: 
remote: Found 1 objects to send. (10 ms)        
Receiving objects: 100% (1/1)
Receiving objects: 100% (1/1), 184 bytes | 184.00 KiB/s, done.
From https://dev.azure.com/ORG/PROJECT/_git/REPO
 * [new ref]         edb49c62b56ceb393542ebb413c58b39130fd07e -> origin/edb49c62b56ceb393542ebb413c58b39130fd07e
git --config-env=http.extraheader=env_var_http.extraheader fetch --force --tags --prune --prune-tags --progress --no-recurse-submodules origin --depth=1 --filter=tree:0 +edb49c62b56ceb393542ebb413c58b39130fd07e
remote: Azure Repos        
remote: 
remote: Found 0 objects to send. (19 ms)        
From https://dev.azure.com/ORG/PROJECT/_git/REPO
 * branch            edb49c62b56ceb393542ebb413c58b39130fd07e -> FETCH_HEAD
git checkout --progress --force refs/remotes/origin/edb49c62b56ceb393542ebb413c58b39130fd07e
fatal: could not read Password for 'https://ORG@dev.azure.com': terminal prompts disabled
fatal: could not fetch 7f12c57ea65125ee8da773d3de56ea12bc338daa from promisor remote
##[warning]Git checkout failed on shallow repository, this might because of git fetch with depth '1' doesn't include the checkout commit 'refs/remotes/origin/edb49c62b56ceb393542ebb413c58b39130fd07e'. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=829603)
##[error]Git checkout failed with exit code: 128
Finishing: Checkout REPO@main to s
```

**Comment by afscrome** [comment]
Setting `AGENT_USE_FETCH_FILTER_IN_CHECKOUT_TASK ` no longer seems to be required, but the issue still occurs.

I believe the problem is the final checkout call is missing credentials (presumably `--config-env=http.extraheader=env_var_http.extraheader`)

```
git checkout --progress --force refs/remotes/origin/edb49c62b56ceb393542ebb413c58b39130fd07e
```

**Comment by ivanduplenskikh** [comment]
Hi @afscrome, thanks for reporting.
I have reproduced the issue.
It arises when the checkout command is executed on an agent that either lacks access to the credential manager or does not have the necessary credentials for the requested repository in credential-manager.
I will keep you updated as I make further progress.

**Comment by ivanduplenskikh** [comment]
@afscrome, I've linked the PR that addresses the issue and is currently under review.

**Comment by ivanduplenskikh** [comment]
@afscrome, could you kindly verify if the issue with your pipeline has been addressed?

**Comment by richshadman** [comment]
We are still seeing this issue @afscrome. 

We are on agent version 3.246.0. Our situation is a little bit complex - we are doing comparisons between source and target commits during a PR to determine impacted projects for testing purposes.

To do this we first check out the target branch, get the commit id, then the source branch, do a comparison between them and get a file list to check against our projects. We have a very large history and therefore are taking advantage of the support for fetch filtering using blob:none to avoid the lengthy download.

When we get to checking out the source commit, if there are blobs that are not already present in the local repository, there is a fetch to the promisor remote (origin) that i believe is still missing the credentials. Please see the screenshot below:

![Image](https://github.com/user-attachments/assets/0d8f30aa-ae93-4b01-9a6c-2cce1b3452a6)

This doesn't happen all the time, only when there is gap that has to be fetched. In these cases having the developer rebase onto the target usually resolves the issue (avoiding the promisor remote fetch). We have also added powershell script to perform some of the git commands directly which seems to work with the persisted credentials just fine.

**Comment by ivanduplenskikh** [comment]
@richshadman, could you please provide a reproducible sample of the pipeline?

**Comment by asyncmeow** [comment]
Hello!
I'm having a similar issue with a fairly minimal pipeline - just a few steps split over two jobs and two stages (one job per stage):
```
# .azure-pipelines.yml
trigger:
  - '*'

pool:
  name: # [azure devops managed pool name]

stages:
  - stage: DatadogMetadata
    displayName: "Upload Git metadata to Datadog"
    jobs:
      - job: DatadogMetadata
        displayName: "Upload Git metadata to Datadog"
        steps:
          - task: Bash@3
            displayName: "Upload metadata"
            env:
              DATADOG_API_KEY: $(ddApiKey)
            inputs:
              targetType: 'inline'
              script: |
                curl -L --fail "https://github.com/DataDog/datadog-ci/releases/latest/download/datadog-ci_linux-x64" --output "/usr/local/bin/datadog-ci"
                chmod +x /usr/local/bin/datadog-ci
                /usr/local/bin/datadog-ci git-metadata upload

  - template: ./azure-pipelines-ci/service-deploy.yml
    parameters:
      environment: staging
      api: customer
---
# ./azure-pipelines-ci/service-deploy.yml
parameters:
  environment: ''
  api: ''

stages:
  - stage: Deploy_${{ parameters.environment }}_${{ parameters.api }}
    displayName: "Deploy - ${{ parameters.environment }} ${{ parameters.api }}"
    jobs:
      - job: Deploy
        displayName: "Deploy"
        steps:
          - task: Docker@2
            displayName: 'Login to Container Registry'
            inputs:
              containerRegistry: # container registry name here
              command: 'login'
          - task: DotNetCoreCLI@2
            inputs:
              command: 'publish'
              publishWebProjects: false
              arguments: '--os linux --arch x64 /t:PublishContainer ./APIs/${{ parameters.api }}Api/${{ parameters.api }}Api.csproj'
              zipAfterPublish: false
              modifyOutputPath: false
```
 
yet, when Azure DevOps tries to checkout the repository:
```
git --config-env=http.extraheader=env_var_http.extraheader fetch --force --tags --prune --prune-tags --progress --no-recurse-submodules origin   +cb3c27436193b148fd065305592de49f455fbd40
From https://dev.azure.com/[OrganizationName]/[ProjectName]/_git/[RepoName]
 * branch            cb3c27436193b148fd065305592de49f455fbd40 -> FETCH_HEAD
git checkout --progress --force cb3c27436193b148fd065305592de49f455fbd40
fatal: could not read Password for 'https://[OrganizationName]@dev.azure.com': terminal prompts disabled
fatal: could not fetch a8cde9df9bf2b3de9d35a24bb36518fa58b0651d from promisor remote
##[error]Git checkout failed with exit code: 128
Finishing: Checkout [RepoName]@main to s
```

This is with shallow fetch disabled as well, which I tried after getting this error:
```
##[warning]Git checkout failed on shallow repository, this might because of git fetch with depth '1' doesn't include the checkout commit 'refs/remotes/origin/cb3c27436193b148fd065305592de49f455fbd40'. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=829603)
```

Any ideas how I could fix this?

**Comment by jeremytbrun** [comment]
I'm getting the same behavior as above.

Basic .NET pipeline with no explicit checkout task.

```
trigger:
- master

pool:
  name: mySelfHostedPool

variables:
  solution: '**/*.sln'
  buildPlatform: 'Any CPU'
  buildConfiguration: 'Release'

steps:
- task: NuGetToolInstaller@1
  displayName: Nu Get Installer 6.x
  inputs:
    versionSpec: 6.x
    
- task: NuGetCommand@2
  inputs:
    command: 'restore'
    feedsToUse: config
    nugetConfigPath: ./nuget.config
    restoreSolution: '$(solution)'
    noCache: true
  continueOnError: true

- task: UseDotNet@2
  displayName: 'Install .Net 8 SDK'
  inputs:
    packageType: 'sdk'
    version: '8.0.x'

- task: DotNetCoreCLI@2
  displayName: '.NET Build'
  inputs:
    projects: '$(solution)'
    command: publish
    publishWebProjects: true
    arguments: '--configuration release --output $(build.artifactstagingdirectory)'
    zipAfterPublish: true
    modifyOutputPath: true

- task: FileTransform@2
  displayName: Update Connection String
  inputs:
    folderPath: '$(Build.ArtifactStagingDirectory)/**/*.zip' # string. Required. Package or folder. Default: $(System.DefaultWorkingDirectory)/**/*.zip.
    jsonTargetFiles: 'appSettings.json' 

- task: PublishBuildArtifacts@1
  displayName: Publish build package for API
  inputs:
    PathtoPublish: '$(build.artifactStagingDirectory)/s.zip'
    ArtifactName: 'drop'
  condition: succeededOrFailed()
```

```
From https://dev.azure.com/org/project/_git/repo
 * branch            581fcc1ee190729f3ebef5c0d649af4d1cceefe0 -> FETCH_HEAD
git checkout --progress --force refs/remotes/origin/581fcc1ee190729f3ebef5c0d649af4d1cceefe0
fatal: Cannot prompt because terminal prompts have been disabled.
fatal: Cannot prompt because terminal prompts have been disabled.
fatal: could not read Password for 'https://org@dev.azure.com/org/project/_git/repo': terminal prompts disabled
fatal: could not fetch 076533d2ac5243fb1134ff04629c8566af5a4cf8 from promisor remote
##[warning]Git checkout failed on shallow repository, this might because of git fetch with depth '1' doesn't include the checkout commit 'refs/remotes/origin/581fcc1ee190729f3ebef5c0d649af4d1cceefe0'. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=829603)
##[error]Git checkout failed with exit code: 128
```

**Comment by jeremytbrun** [comment]
@ivanduplenskikh

**Comment by jeremytbrun** [comment]
> Hello! I'm having a similar issue with a fairly minimal pipeline - just a few steps split over two jobs and two stages (one job per stage):
> 
> ```
> # .azure-pipelines.yml
> trigger:
>   - '*'
> 
> pool:
>   name: # [azure devops managed pool name]
> 
> stages:
>   - stage: DatadogMetadata
>     displayName: "Upload Git metadata to Datadog"
>     jobs:
>       - job: DatadogMetadata
>         displayName: "Upload Git metadata to Datadog"
>         steps:
>           - task: Bash@3
>             displayName: "Upload metadata"
>             env:
>               DATADOG_API_KEY: $(ddApiKey)
>             inputs:
>               targetType: 'inline'
>               script: |
>                 curl -L --fail "https://github.com/DataDog/datadog-ci/releases/latest/download/datadog-ci_linux-x64" --output "/usr/local/bin/datadog-ci"
>                 chmod +x /usr/local/bin/datadog-ci
>                 /usr/local/bin/datadog-ci git-metadata upload
> 
>   - template: ./azure-pipelines-ci/service-deploy.yml
>     parameters:
>       environment: staging
>       api: customer
> ---
> # ./azure-pipelines-ci/service-deploy.yml
> parameters:
>   environment: ''
>   api: ''
> 
> stages:
>   - stage: Deploy_${{ parameters.environment }}_${{ parameters.api }}
>     displayName: "Deploy - ${{ parameters.environment }} ${{ parameters.api }}"
>     jobs:
>       - job: Deploy
>         displayName: "Deploy"
>         steps:
>           - task: Docker@2
>             displayName: 'Login to Container Registry'
>             inputs:
>               containerRegistry: # container registry name here
>               command: 'login'
>           - task: DotNetCoreCLI@2
>             inputs:
>               command: 'publish'
>               publishWebProjects: false
>               arguments: '--os linux --arch x64 /t:PublishContainer ./APIs/${{ parameters.api }}Api/${{ parameters.api }}Api.csproj'
>               zipAfterPublish: false
>               modifyOutputPath: false
> ```
> 
> yet, when Azure DevOps tries to checkout the repository:
> 
> ```
> git --config-env=http.extraheader=env_var_http.extraheader fetch --force --tags --prune --prune-tags --progress --no-recurse-submodules origin   +cb3c27436193b148fd065305592de49f455fbd40
> From https://dev.azure.com/[OrganizationName]/[ProjectName]/_git/[RepoName]
>  * branch            cb3c27436193b148fd065305592de49f455fbd40 -> FETCH_HEAD
> git checkout --progress --force cb3c27436193b148fd065305592de49f455fbd40
> fatal: could not read Password for 'https://[OrganizationName]@dev.azure.com': terminal prompts disabled
> fatal: could not fetch a8cde9df9bf2b3de9d35a24bb36518fa58b0651d from promisor remote
> ##[error]Git checkout failed with exit code: 128
> Finishing: Checkout [RepoName]@main to s
> ```
> 
> This is with shallow fetch disabled as well, which I tried after getting this error:
> 
> ```
> ##[warning]Git checkout failed on shallow repository, this might because of git fetch with depth '1' doesn't include the checkout commit 'refs/remotes/origin/cb3c27436193b148fd065305592de49f455fbd40'. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=829603)
> ```
> 
> Any ideas how I could fix this?

Have you made any progress in figuring this out? I'm still experiencing it and haven't yet been able to get to the bottom of it.

**Comment by asyncmeow** [comment]
Not yet... I put the project on hold at work and I'm working on other stuff until it's figured out. I've been trying to get my boss to escalate it with a contact our org has at Microsoft but he has yet to send an email he's had drafted for over a week now...On Feb 23, 2025 16:06, Jeremy Brun ***@***.***> wrote:
  
Hello! I'm having a similar issue with a fairly minimal pipeline - just a few steps split over two jobs and two stages (one job per stage):
# .azure-pipelines.yml
trigger:
  - '*'

pool:
  name: # [azure devops managed pool name]

stages:
  - stage: DatadogMetadata
    displayName: "Upload Git metadata to Datadog"
    jobs:
      - job: DatadogMetadata
        displayName: "Upload Git metadata to Datadog"
        steps:
          - task: ***@***.***
            displayName: "Upload metadata"
            env:
              DATADOG_API_KEY: $(ddApiKey)
            inputs:
              targetType: 'inline'
              script: |
                curl -L --fail "https://github.com/DataDog/datadog-ci/releases/latest/download/datadog-ci_linux-x64" --output "/usr/local/bin/datadog-ci"
                chmod +x /usr/local/bin/datadog-ci
                /usr/local/bin/datadog-ci git-metadata upload

  - template: ./azure-pipelines-ci/service-deploy.yml
    parameters:
      environment: staging
      api: customer
---
# ./azure-pipelines-ci/service-deploy.yml
parameters:
  environment: ''
  api: ''

stages:
  - stage: Deploy_${{ parameters.environment }}_${{ parameters.api }}
    displayName: "Deploy - ${{ parameters.environment }} ${{ parameters.api }}"
    jobs:
      - job: Deploy
        displayName: "Deploy"
        steps:
          - task: ***@***.***
            displayName: 'Login to Container Registry'
            inputs:
              containerRegistry: # container registry name here
              command: 'login'
          - task: ***@***.***
            inputs:
              command: 'publish'
              publishWebProjects: false
              arguments: '--os linux --arch x64 /t:PublishContainer ./APIs/${{ parameters.api }}Api/${{ parameters.api }}Api.csproj'
              zipAfterPublish: false
              modifyOutputPath: false

yet, when Azure DevOps tries to checkout the repository:
git --config-env=http.extraheader=env_var_http.extraheader fetch --force --tags --prune --prune-tags --progress --no-recurse-submodules origin   +cb3c27436193b148fd065305592de49f455fbd40
From https://dev.azure.com/[OrganizationName]/[ProjectName]/_git/[RepoName]
 * branch            cb3c27436193b148fd065305592de49f455fbd40 -> FETCH_HEAD
git checkout --progress --force cb3c27436193b148fd065305592de49f455fbd40
fatal: could not read Password for ***@***.***': terminal prompts disabled
fatal: could not fetch a8cde9df9bf2b3de9d35a24bb36518fa58b0651d from promisor remote
##[error]Git checkout failed with exit code: 128
Finishing: Checkout ***@***.*** to s

This is with shallow fetch disabled as well, which I tried after getting this error:
##[warning]Git checkout failed on shallow repository, this might because of git fetch with depth '1' doesn't include the checkout commit 'refs/remotes/origin/cb3c27436193b148fd065305592de49f455fbd40'. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=829603)

Any ideas how I could fix this?

Have you made any progress in figuring this out? I'm still experiencing it and haven't yet been able to get to the bottom of it.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>



jeremytbrun left a comment (microsoft/azure-pipelines-agent#4860)

Hello! I'm having a similar issue with a fairly minimal pipeline - just a few steps split over two jobs and two stages (one job per stage):
# .azure-pipelines.yml
trigger:
  - '*'

pool:
  name: # [azure devops managed pool name]

stages:
  - stage: DatadogMetadata
    displayName: "Upload Git metadata to Datadog"
    jobs:
      - job: DatadogMetadata
        displayName: "Upload Git metadata to Datadog"
        steps:
          - task: ***@***.***
            displayName: "Upload metadata"
            env:
              DATADOG_API_KEY: $(ddApiKey)
            inputs:
              targetType: 'inline'
              script: |
                curl -L --fail "https://github.com/DataDog/datadog-ci/releases/latest/download/datadog-ci_linux-x64" --output "/usr/local/bin/datadog-ci"
                chmod +x /usr/local/bin/datadog-ci
                /usr/local/bin/datadog-ci git-metadata upload

  - template: ./azure-pipelines-ci/service-deploy.yml
    parameters:
      environment: staging
      api: customer
---
# ./azure-pipelines-ci/service-deploy.yml
parameters:
  environment: ''
  api: ''

stages:
  - stage: Deploy_${{ parameters.environment }}_${{ parameters.api }}
    displayName: "Deploy - ${{ parameters.environment }} ${{ parameters.api }}"
    jobs:
      - job: Deploy
        displayName: "Deploy"
        steps:
          - task: ***@***.***
            displayName: 'Login to Container Registry'
            inputs:
              containerRegistry: # container registry name here
              command: 'login'
          - task: ***@***.***
            inputs:
              command: 'publish'
              publishWebProjects: false
              arguments: '--os linux --arch x64 /t:PublishContainer ./APIs/${{ parameters.api }}Api/${{ parameters.api }}Api.csproj'
              zipAfterPublish: false
              modifyOutputPath: false

yet, when Azure DevOps tries to checkout the repository:
git --config-env=http.extraheader=env_var_http.extraheader fetch --force --tags --prune --prune-tags --progress --no-recurse-submodules origin   +cb3c27436193b148fd065305592de49f455fbd40
From https://dev.azure.com/[OrganizationName]/[ProjectName]/_git/[RepoName]
 * branch            cb3c27436193b148fd065305592de49f455fbd40 -> FETCH_HEAD
git checkout --progress --force cb3c27436193b148fd065305592de49f455fbd40
fatal: could not read Password for ***@***.***': terminal prompts disabled
fatal: could not fetch a8cde9df9bf2b3de9d35a24bb36518fa58b0651d from promisor remote
##[error]Git checkout failed with exit code: 128
Finishing: Checkout ***@***.*** to s

This is with shallow fetch disabled as well, which I tried after getting this error:
##[warning]Git checkout failed on shallow repository, this might because of git fetch with depth '1' doesn't include the checkout commit 'refs/remotes/origin/cb3c27436193b148fd065305592de49f455fbd40'. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=829603)

Any ideas how I could fix this?

Have you made any progress in figuring this out? I'm still experiencing it and haven't yet been able to get to the bottom of it.

**Comment by jeremytbrun** [comment]
@asyncmeow thanks. I have a ticket open with Microsoft but so far nothing helpful has come of it.

Here are my observations so far.

- It's a consistent but still intermittent issue with only 2 or 3 of our repos in our org. The significant majority of our repos don't have the issue.
- I have not yet seen the issue on an MS hosted agent. It's only occuring on our own self-hosted agents.
- It occurs even on brand new self-hosted agents.

**Comment by richshadman** [comment]
> [@afscrome](https://github.com/afscrome), I'm closing this issue for the time being. Feel free to re-open this issue if you have any additional questions.
> 
> [@richshadman](https://github.com/richshadman), could you please clarify if you are using [the checkout task](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-checkout?view=azure-pipelines) or if you are executing the checkout by custom script?

@afscrome It took me a while but I finally have a project/repo that I can reproduce the issue for you. It was pretty simple:

1. Create a new repository, main branch, initialized with a "Readme".
2. Create a branch off of that repository, Edit the readme, add a new line of arbitrary text.
3. Add a yaml pipeline to this repository with the following:
``` yaml
name: Checkout Issue Repro

trigger: none

pool:
  vmImage: 'windows-latest'

jobs:
- job: CheckoutTest
  steps:
  - checkout: git://GitCheckoutIssue/GitCheckoutIssue@refs/heads/main
    fetchTags: false
    fetchFilter: blob:none
    displayName: Checkout Test - Main

  - checkout: self
    fetchTags: false
    displayName: Checkout Test - Self
```
4. Commit / Push changes
5. Switch back to main, edit readme, same line, different text
6. Commit / Push changes
7. Run the pipeline

See the results:

![Image](https://github.com/user-attachments/assets/0c3fbe3c-355b-4122-8422-3c2a2d3dbf0c)
 
I can grant you access to the project/repo as guest access if needed - please reach out. Either way, the steps above should reproduce the issue for you. 

Bottom line - credentials are NOT being sent for promisor fetches to fill in the gaps. Please let me know if you want to create a new ticket for this.

**Comment by richshadman** [comment]
A few additional comments, hope this explains things a bit to others seeing this issue intermittently:

1. It won't occur on hosted agents without a workflow like the one I mentioned above as you need an environment where the git repo is already initialized with a fetch filter (partial clone) - my example explicitly uses blob:none
2. After that you need a checkout that needs commits already not pulled to the checked out repository to initiate a "promisor" fetch (read up on how git handles partial clones to understand this)
3. In my example, I have the 2 checkouts because our real world example is doing a diff between source and target during pull requests to find changed files. We *must* do a partial clone with the fetch filter because of the size of our history - slows things down tremendously otherwise.

**Comment by richshadman** [comment]
I went ahead and opened a new bug for the issue, hoping it gets better traction. This has an impact weekly if not daily.
